src/nupic/algorithms/Cells4.hpp

```diff
@@ -60,1092 +61,1182 @@
-  //
-  // ROUTINES FOR PERSISTENCE
-  //
-  //----------------------------------------------------------------------
-  //----------------------------------------------------------------------
-
-  /**
-   * TODO: compute, rather than writing to a buffer.
-   * TODO: move persistence to binary, faster and easier to compute expecte
-   * size.
-   */
-  UInt persistentSize() const {
-    // TODO: this won't scale!
-    std::stringstream tmp;
-    this->save(tmp);
-    return tmp.str().size();
-  }
-
-  //----------------------------------------------------------------------
-  /**
-   * Write the state to a proto or file
-   */
-  using Serializable::write;
-  virtual void write(Cells4Proto::Builder &proto) const override;
-
-  //----------------------------------------------------------------------
-  /**
-   * Read the state into a proto or file
-   */
-  using Serializable::read;
-  virtual void read(Cells4Proto::Reader &proto) override;
-
-  //----------------------------------------------------------------------
-  /**
-   * Save the state to the given file
-   */
-  void saveToFile(std::string filePath) const;
-
-  //----------------------------------------------------------------------
-  /**
-   * Load the state from the given file
-   */
-  void loadFromFile(std::string filePath);
-
-  //----------------------------------------------------------------------
-  void save(std::ostream &outStream) const;
-
-  //-----------------------------------------------------------------------
-  /**
-   * Need to load and re-propagate activities so that we can really persist
-   * at any point, load back and resume inference at exactly the same point.
-   */
-  void load(std::istream &inStream);
-
-  //-----------------------------------------------------------------------
-  void print(std::ostream &outStream) const;
-
-  //----------------------------------------------------------------------
-  //----------------------------------------------------------------------
-  //
-  // MISC SUPPORT AND DEBUGGING ROUTINES
-  //
-  //----------------------------------------------------------------------
-  //----------------------------------------------------------------------
-
-  // Set the Cell class segment order
-  void setCellSegmentOrder(bool matchPythonOrder);
-
-  //----------------------------------------------------------------------
-  /**
-   * Used in unit tests and debugging.
-   */
-  void addNewSegment(UInt colIdx, UInt cellIdxInCol, bool sequenceSegmentFlag,
-                     const std::vector<std::pair<UInt, UInt>> &extSynapses);
-
-  void updateSegment(UInt colIdx, UInt cellIdxInCol, UInt segIdx,
-                     const std::vector<std::pair<UInt, UInt>> &extSynapses);
-
-  //-----------------------------------------------------------------------
-  /**
-   * Rebalances and rebuilds internal structures for faster computing
-   *
-   */
-  void _rebalance();
-  
```

Comment:
David, please figure how your IDE (auto)transforms code. I think there is a code-style for c++ nupic.